### PR TITLE
Created the plotfasta function and loadkmers function

### PR DESCRIPTION
**Pull request by Thohil**
Created a function to plot fasta data using the kmers
Lazar created a function to load in kmers from the .model file

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -2,3 +2,9 @@ name = "NanoporeMultiEventalign"
 uuid = "c438b98d-67fd-45c3-9996-84dfdeb636ca"
 authors = ["JokingHero <kornellabun@gmail.com>"]
 version = "0.1.0"
+
+[deps]
+BioSequences = "7e6ae17a-c86d-528c-b3b9-7f778a29fe59"
+FASTX = "c2308a5c-f048-11e8-3e8a-31650f418d12"
+Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
+PyPlot = "d330b81b-6aea-500a-939a-2ce795aea3ee"
```
Comment (JokingHero) [inline Project.toml]: PyPlot not needed

File: src/NanoporeMultiEventalign.jl [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
 
 module NanoporeMultiEventalign
+using BioSequences
+using FASTX
+using Plots
 
-include("utils.jl")
 include("persistence.jl")
 include("dtw.jl")
+include("utils.jl")
```
Comment (JokingHero) [inline src/NanoporeMultiEventalign.jl]: this should be at the top

File: src/NanoporeMultiEventalign.jl [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
+
+#plotdata("test/sample_data/reference.fa")
```
Comment (JokingHero) [inline src/NanoporeMultiEventalign.jl]: should not be here

File: src/persistence.jl [hunk 4]
```diff
@@ -1 +1,27 @@
...
+    fastaList = []
+    for r in f
+          push!(fastaList, FASTA.sequence(r))
+    end
+    close(f)
+    return fastaList
+end
+
+# Makes kmers dictionary that stores mean and stdvt for each kmer.
+# Access syntax: kmers["GCTGT"][2] gets stdvt of the kmer
```
Comment (JokingHero) [inline src/persistence.jl]: you can put this into documentation with 

"

"

File: src/persistence.jl [hunk 5]
```diff
@@ -1 +1,27 @@
...
+function loadkmers()
+    model_path = "models/r9.4_70bps.u_to_t_rna.5mer.template.model"
```
Comment (JokingHero) [inline src/persistence.jl]: should be parameter with default

File: src/utils.jl [hunk 6]
```diff
@@ -0,0 +1,15 @@
+
+
+function plotdata(path::String)
+    kmers = loadkmers()
+    data = loadfasta(path)
+    dataString = convert(String, data[1])
```
Comment (JokingHero) [inline src/utils.jl]: keep the naming conventions consistent with julia style

File: src/utils.jl [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    plotList = []
+    for i in 1:(length(dataString)-4)
```
Comment (JokingHero) [inline src/utils.jl]: assumes the size of the kmer

File: src/utils.jl [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+        kmerKey = dataString[i:(i+4)]
+        #collects all the kmer values of the data into a lits
+        push!(plotList,parse(Float64,kmers[kmerKey][1]))
```
Comment (JokingHero) [inline src/utils.jl]: keep spaces etc

File: src/utils.jl [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    end
+    plot(1:(length(plotList)),plotList)
+    png("plot")#saves the plot as a png named plot
```
Comment (JokingHero) [inline src/utils.jl]: plot path should be input

File: src/utils.jl [hunk 10]
```diff
@@ -0,0 +1,15 @@
+
+
+function plotdata(path::String)
```
Comment (JokingHero) [inline src/utils.jl]: instead make funciton that processes fasta into kmer mean points

**Review by JokingHero — CHANGES_REQUESTED** [review]

File: src/NanoporeMultiEventalign.jl [hunk 11]
```diff
@@ -15,6 +15,7 @@ include("dtw.jl")
 
 
 export dtw
-export loadfasta, loadkmers, fasta_to_kmer_valus
+export loadnanoporefast5, loadfasta, loadkmers, fasta_to_kmer_valus
```
Comment (JokingHero) [inline src/NanoporeMultiEventalign.jl]: should be values?
fasta_to_model maybe? 

After you fix this try merging this branch to the main, fix conflicts beforehand.

**Review by JokingHero — APPROVED** [review]
Adress my comment and merge this to the main.

**Comment by Thohil** [comment]
It says "Only those with write access to this repository can merge pull requests." so i am not sure if i have permission to merge the changes into main
image

**Comment by JokingHero** [comment]
You have not accepted the invitation to be the contributor, its waiting for some time now.

**Comment by Thohil** [comment]
I found the invitation, however github invitations expire after 7 days so i can't use it. is it possible for you to resend the invitation?

**Pull request merged**